client/src/hooks/useFetch.js

```diff
@@ -1,19 +1,25 @@
 import React from 'react';
 
 function useFetch(url) {
-  const [user, setUser] = React.useState(null);
+  const [data, setData] = React.useState([]);
```

Comment:
`data` is not always an array. 
In the `api/auth` case, it's an object `{}`. 
I would suggest reverting this change and set the initial value to `null` to be independent of the result type.